src/EFCore.PG/Metadata/PostgresRange.cs

```diff
@@ -9,17 +9,47 @@
 
 namespace Npgsql.EntityFrameworkCore.PostgreSQL.Metadata
 {
+    /// <summary>
+    /// Represents the metadata for a PostgreSQL range.
+    /// </summary>
+    [PublicAPI]
     public class PostgresRange
     {
         [NotNull] readonly IAnnotatable _annotatable;
         [NotNull] readonly string _annotationName;
 
+        /// <summary>
+        /// Creates a <see cref="PostgresRange"/>.
+        /// </summary>
+        /// <param name="annotatable">The annotatable to search for the annotation.</param>
+        /// <param name="annotationName">The annotation name to search for in the annotatable.</param>
+        /// <exception cref="ArgumentNullException"><paramref name="annotatable"/></exception>
```

Comment:
I don't think these exception xmldocs have much value (not sure I've actually seen them specified in other projects etc.). I'm especially skeptical of documenting each and every internal method, it seems to bloat the source for very little gain...

But let's move on...

Reply:
FWIW, they show up in the on-hover documentation, which is nice for a quick check on whether or not a null parameter is going to throw.